src/main/java/net/snowflake/client/core/SessionUtilExternalBrowser.java

```diff
@@ -265,6 +269,7 @@ private String generateProofKey() {
   void authenticate() throws SFException, SnowflakeSQLException {
     ServerSocket ssocket = this.getServerSocket();
     try {
+      ssocket.setSoTimeout(getBrowserResponseTimeout());
```

Comment:
is it a correct timeout? can we add (unit) test to verify that we are waiting at least for configured timeout and then fail the autentication process? what exception is thrown then?

Reply:
I'm still working on if we can write a unit test for this. For now I've been manually testing by trying to connect with an external browser, then closing the browser window when it pops up, and with the connection property set, I see that an error gets thrown after the correct number of seconds. The error we get is `net.snowflake.client.jdbc.SnowflakeSQLLoggedException: JDBC driver encountered communication error. Message: Accept timed out.`
Before the driver would just hang forever.

Reply:
if you have a manual test please add it also to the PR or let us know if you are using any of the existing one

to write an unit test maybe you will need to extract some package scope methods in this class - please do not hesitate to do it